app/src/main/java/com/example/instagram/fragments/DetailFragment.java

```diff
@@ -0,0 +1,289 @@
+                }
+            }
+        });
+
+        // Profile image leads to profile fragment
+        ivProfileImage.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    goToProfile();
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+
+        // Username leads to profile fragment
+        tvUsername.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    goToProfile();
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+
+        // Post input comment when comment button is clicked
+        btnComment.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String comment = etComment.getText().toString();
+                post.postComment(comment, getActivity());
+                etComment.setText("");
+                queryComments();
+            }
+        });
+
+        // Allow user to like with double click
+        ivImage.setOnClickListener(new DoubleClickListener() {
+            @Override
+            public void onSingleClick(View v) {
+            }
+
+            @Override
+            public void onDoubleClick(View v) {
+                try {
+                    post.attemptToLike(ParseUser.getCurrentUser());
+                    updateLike();
+                    updateNumLike();
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    private void updateSaved() throws ParseException {
+        if (user.postSaved(post)) {
+            ivShare.setImageResource(R.drawable.nav_inbox_paper_plane);
+        } else {
+            ivShare.setImageResource(R.drawable.direct);
+        }
+    }
+
+    private void updateLike() throws ParseException {
+        if (post.userLiked(ParseUser.getCurrentUser())) {
+            ivLike.setImageResource(R.drawable.ufi_heart_active);
+        } else {
+            ivLike.setImageResource(R.drawable.ufi_heart);
+        }
+    }
+
+    private void updateNumLike() throws ParseException {
+        int numLike = post.getNumLike();
+        if (numLike > 1) {
+            tvNumLike.setText("" + numLike + " likes");
+        } else {
+            tvNumLike.setText("" + numLike + " like");
+        }
+    }
+
+    private void queryComments() {
+        Log.i(TAG, "Start querying for new comments");
+
+        // Set up query
+        ParseQuery<Comment> query = ParseQuery.getQuery(Comment.class);
+        query.include(Post.KEY_USER);
+        query.whereContains("post", post.getObjectId());
+        query.addDescendingOrder("createdAt");
```

Comment:
Would be good to use keys like "KEY_POST" from the model class instead of the raw strings